### ISSUE TITLE
Order on Multiple Fields

### ISSUE THREAD

**Issue by jjallen37** [litedb-org/LiteDB#1409]
I am trying to set up a query in v5 Beta, but am running into an issue when trying to order on multiple fields, which does not appear to be supported at the moment. I receive the error message "ORDER BY already defined in this query builder". I was hoping to use a syntax similar to Linq's OrderBy/ThenBy. Is this something that LiteDB would be able to support?

What I Attempted:
![image](https://user-images.githubusercontent.com/2042622/72829354-9b5d8580-3c4c-11ea-8122-63917d405fc2.png)

**Comment by jjallen37** [comment]
Assuming this was not something that would be quickly available in LiteDB, I've been trying some other approaches to sort things without pulling too much into memory. We're updating to v5 because the addition of "Select" has drastically improved our performance on more complicated objects. I think I can accomplish what I want if I get the sorted list of id's then use that to query my data, but I don't know how to do a Join-type operation here.

tldr; Can I efficiently retrieve data if I have a list of id's?
![image](https://user-images.githubusercontent.com/2042622/72830435-ba5d1700-3c4e-11ea-905a-a08ebd5c7d66.png)

**Comment by lbnascimento** [comment]
Hello!

Unfortunately multi-field sorting is not supported yet (it is a planned feature for LiteDb v5.1). A multi-field sorting could be achieved by using an array or document containing the fields as the sorting expression, however this would only work if all the fields were to be sorted in the same order (ascending or descending).

Joins are not avaliable in LiteDB. If memory usage is a concern, I recommend first querying for a sorted list of ids, as you suggested, and then iterating through this list querying by id and selecting the fields you need. This should not cause any performance issues, given that an index would be used in the queries.

**Comment by jjallen37** [comment]
@lbnascimento Thanks for the quick reply! I was able to get things working via the second implementation, where I iterated through the list of ids and queried for each one. This is currently fast enough to meet my needs, but I haven't done extensive testing for performance it.
![image](https://user-images.githubusercontent.com/2042622/72989215-da5e1900-3dbb-11ea-8ab4-c5ddf9ce1b7f.png)


As a general question though, I would assume that doing a query for each id would be more expensive than one query that could return all of the objects. Is this correct?

**Comment by lbnascimento** [comment]
@jjallen37 You are correct, _some_ performance loss is to be expected, however querying by id should be fast enough for most usages.

**Comment by Haapavuo** [comment]
Hopefully a ThenBy() method will arrive to LiteDB soon. Really looking forward to it.

**Comment by JKamsker** [comment]
This is the same problem as #1330, which is fixed on the `dev` branch. `dev` ships as the LiteDB 6.x prereleases.

Please try the latest prerelease (currently **6.0.0-prerelease.81**):

```
dotnet add package LiteDB --prerelease
```

If the problem still happens on the prerelease, comment here with the version you used and a repro, and we'll reopen the issue.
